src/App.tsx

```diff
@@ -14,37 +17,132 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum ListOperation {
+  SORT_ALPHABET = 'Sort alphabetically',
+  SORT_LENGTH = 'Sort by length',
+  REVERSE = 'Reverse',
+  RESET = 'Reset',
+}
+
+function styleClass(btn: ListOperation) {
+  switch (btn) {
+    case ListOperation.SORT_ALPHABET:
+      return 'is-info';
+    case ListOperation.SORT_LENGTH:
+      return 'is-success';
+    case ListOperation.REVERSE:
+      return 'is-warning';
+    case ListOperation.RESET:
+      return 'is-danger';
+    default:
+      return '';
+  }
+}
+
+type Sort = {
+  typeField: string;
+  isReversed: boolean;
+};
+
+function sortGoods(goods: string[], { typeField, isReversed }: Sort) {
+  const changedGoods = [...goods];
+
+  switch (typeField) {
+    case ListOperation.SORT_ALPHABET:
+      changedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case ListOperation.SORT_LENGTH:
+      changedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    changedGoods.reverse();
+  }
+
+  return changedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sort, setSort] = useState({ typeField: '', isReversed: false });
```

Comment:
The initial state for `typeField` is an empty string. Consider setting a default operation or handling this case explicitly in the `sortGoods` function to avoid potential issues with sorting logic.